### DISCUSSION TITLE
Some way to override route

### DISCUSSION THREAD

**Discussion by veesahni** [withastro/roadmap#380] [Proposal]
I would like to be able to route things differently than what is on the filesystem

For example, I store content in ```src/blog/[name].mdx``` and ```src/case-studies/[name].mdx``` but I don't want ```/blog/``` or ```/case-studies/``` in the URL.

Solution: Perhaps astro could look for a file in the tree (eg: ```src/blog/astro.conf```) that allows re-rooting this folder up a level.  Or perhaps this could be some sort of routing option (that works on a per-folder basis) available in the main astro config file.

**Comment by delucis** [comment]
One option for this that is available today is to use the `injectRoute` helper in the integrations API: https://docs.astro.build/en/reference/integrations-reference/#injectroute-option

It allows you to inject pages that live anywhere in your project to a path of your choice. It's not the nicest DX but does work!

**Comment by veesahni** [comment]
@delucis appreciate the pointer! It got me thinking more about dynamic routes and I found a solution that works for my use case:

With the following ```[slug].astro``` into ```src/pages```,  I can now store mdx entry files in any structure in ```src/content``` and the mdx files are in control of their route.

```
---
export async function getStaticPaths(){
  const paths = await Astro.glob('../content/**/*.mdx')

  return paths.map(p => (
    {
      params: {slug: p.frontmatter.slug},
      props: p
    }
  ))
}
const { Content } = Astro.props
---
<Content />
```

**Reply by veesahni** [reply]
I also noticed that conflicting routes don't currently raise an error - I've [opened a bug report for that](https://github.com/withastro/astro/issues/5340)

**Comment by florian-lefebvre** [comment]
Closing because this is outdated
